source/api/model/data-model.md

```diff
@@ -0,0 +1,41 @@
+---
+title: DataModel
+type: api
+group: API
+order: 5
+---
+A class that gives access to the high level grid data structures.
+Usually grid user can define this properties in different places. Inside html through attribute bindings:
+```html
+<q-grid [columns]="userColumns" [rows]="userRows">
+```
+
+Inside html throught component:
+```html
+<q-grid>
+	<q-grid-columns>
+		<q-grid-column key="id"></q-grid-column>
+		<q-grid-column key="name"><q-grid-column>
+	</q-grid-columns>
+</q-grid>
+```
+
+Inside js code throught model:
+```javascript
+const userRows = [];
+const userColumns = [];
+
+gridModel.data({
+ rows: userRows,
+ columns: userColumns
+})
+```
+
+
+Property|Type|Description
+---|---|---
+columns|`array<ColumnModel>`|Set of columns to display.<br />Usually data columns can be setup from 3 places:<br />1. Columns binding property.<br />2. Columns component.<br />3. Grid model data columns property.<br />We can have 3 sources of columns because each column has `key` property,<br />that allows to make a merge. If you have defined columns in javascript and in template<br />with the same key, algorithm will try persist settings from both sources but<br />javascript will have top priority.
+pipe|`array<IPipe>`|Chaing of methods that grid invokes asyncroniuosly anytime refresh is required,<br />see `PipeModel` that contains information when grid demands refreshing.
```

Comment:
typo in pipe property description - Chaing instead of Chain

Reply:
This is just test text, for UI/UX review will change it letter maybe whole thing